src/plugins/barista/tea.rs

```diff
@@ -0,0 +1,447 @@
+            Adjective::FairTrade => f.write_str("fair trade"),
+            Adjective::Homemade => f.write_str("homemade"),
+            Adjective::HomeBrewn => f.write_str("home-brewn"),
+            Adjective::Kangra => f.write_str("Kangra"),
+            Adjective::Irish => f.write_str("Irish-breakfast"),
+            Adjective::English => f.write_str("English-breakfast"),
+            Adjective::Darjeel => f.write_str("Darjeeling"),
+            Adjective::Vanilla => f.write_str("vanilla"),
+            Adjective::Lemongrass => f.write_str("lemongrass"),
+            Adjective::Hibiscus => f.write_str("hibiscus"),
+        }
+    }
+}
+
+const SIZES: &[&str] = &[
+    "large", "small", "medium", "tall", "wide", "big", "100ml", "giant", "tiny",
+];
+const FILLED_WITH: &[&str] = &[
+    "filled with",
+    "of",
+    "stuffed with",
+    "full of",
+    "brimming with",
+];
+
+// Temperature of tea.
+const COLD: &[&str] = &["iced", "cold", "chilled", "ice cold"];
+const HOT: &[&str] = &[
+    "boiling",
+    "scalding",
+    "steaming",
+    "sweltering",
+    "toasty hot",
+];
+
+// It's unfortunate, but the best way to handle this (because there is no const
+// flatten) is manually copying them.
+const WARM_HOT: &[&str] = &[
+    // Warm
+    "lukewarm",
+    "warm",
+    "warmish",
+    "room temperature",
+    // Hot
+    "boiling",
+    "scalding",
+    "steaming",
+    "sweltering",
+    "toasty hot",
+];
+const COLD_HOT: &[&str] = &[
+    // Cold
+    "iced",
+    "cold",
+    "chilled",
+    "ice cold",
+    // Hot
+    "boiling",
+    "scalding",
+    "steaming",
+    "sweltering",
+    "toasty hot",
+];
+const ALL: &[&str] = &[
+    // Cold
+    "iced",
+    "cold",
+    "chilled",
+    "ice cold",
+    // Warm
+    "lukewarm",
+    "warm",
+    "warmish",
+    "room temperature",
+    // Hot
+    "boiling",
+    "scalding",
+    "steaming",
+    "sweltering",
+    "toasty hot",
+];
+
+enum TeaType {
+    Black,
+    Green,
+    MatchaGreen,
+    SenchaGreen,
+    White,
+    Oolong,
+    Puer,
+    Chai,
+    Butter,
+    Christmas,
+    Rooibos,
+    Tulsi,
+    LemonbalmTulsi,
+    Spearmint,
+    Peppermint,
+    ChocolateMint,
+    Mullein,
+    LambsEars,
+    TumericGinger,
+    LemongrassVerbena,
+    Lemongrass,
+    BlackCurrantHibiscus,
+    RoastedDandelionRoot,
+    DandelionLeafAndRoot,
+    Lavender,
+}
+
+impl TeaType {
+    fn vessel_choices(&self) -> &[VesselType] {
+        match *self {
+            TeaType::Black => &[
+                VesselType::Samovar,
+                VesselType::Teapot,
+                VesselType::Mug,
+                VesselType::Teacup,
+            ],
+            TeaType::Green => &[VesselType::Teacup, VesselType::Mug, VesselType::Hohin],
+            TeaType::MatchaGreen => &[
+                VesselType::Hohin,
+                VesselType::Teapot,
+                VesselType::Mug,
+                VesselType::Teacup,
+            ],
```

Comment:
This looks extremely similar to the temperatures stuff. Can you just do the same and create a `Temperature` enum?

Reply:
Done.